src/generator.cpp

```diff
@@ -1373,14 +1377,31 @@ bool Generator::GeneratorImpl::compareVariablesByName(const GeneratorInternalVar
     return realComponent1->name() < realComponent2->name();
 }
 
+bool Generator::GeneratorImpl::isStateVariable(const GeneratorInternalVariablePtr &variable)
+{
+    return variable->mType == GeneratorInternalVariable::Type::STATE;
+}
+
+bool Generator::GeneratorImpl::isConstantOrAlgebraicVariable(const GeneratorInternalVariablePtr &variable)
+{
+    return (variable->mType == GeneratorInternalVariable::Type::CONSTANT)
+           || (variable->mType == GeneratorInternalVariable::Type::COMPUTED_TRUE_CONSTANT)
+           || (variable->mType == GeneratorInternalVariable::Type::COMPUTED_VARIABLE_BASED_CONSTANT)
+           || (variable->mType == GeneratorInternalVariable::Type::ALGEBRAIC);
+}
```

Comment:
What's the difference between these types?  `CONSTANT` and `COMPUTED_TRUE_CONSTANT`? And `COMPUTED_VARIABLE_BASED_CONSTANT` and later you have `COMPUTED_CONSTANT` as well?

Reply:
They are internal types to the generator to determine our different types of constants:
 - `x = 3` --> `CONSTANT`;
 - `y = 3 + 5` --> `COMPUTED_TRUE_CONSTANT`; and
 - `z = a + b` with both `a` and `b` constants --> `COMPUTED_VARIABLE_BASED_CONSTANT`.

However, from the outside (e.g. OpenCOR), we only want to know whether a constant is a "true" constant or a computed constant, hence:
 - internal `CONSTANT` --> external `CONSTANT`;
 - internal `COMPUTED_TRUE_CONSTANT` --> external `COMPUTED_CONSTANT`; and
 - internal `COMPUTED_VARIABLE_BASED_CONSTANT` --> external `COMPUTED_CONSTANT`.

HTH... :)